docs/kpm_sparse_checkout.md

```diff
@@ -0,0 +1,75 @@
+## KPM Sparse-Checkout feature
+
+## 1. Introduction
+kpm which is the package management tool for KCL, does not support sparse-checkout, which means an issue arises when dealing with monorepos, which may contain many KCL packages. We need to get a solution to download specific packages instead of all in a monorepo.
+
+## 2. User Worklow
+- Adding a Subdirectory Dependency via Command Line
+- Specifying a Subdirectory in `kcl.mod`
+- Adding a Subdirectory Dependency via Configuration File 
+
+## 3. Command Line Inteface 
+We will try to keep the command line interface as simple and straightforward as possible for adding a Git repository subdirectory as a dependency. The below steps show how a user can use the feature:
+
+- Adding a Dependency :
+The following command will lead to the addition of a dependency:
+`kcl mod add <repository_url> <subdirectory_path>`
+
+This command will add the specified subdirectory as a dependency in the current KCL project.
+
+- Non-interactive Method :
+In addition to the command line interface, users can specify subdirectory dependencies directly in the `kcl.mod` file, which can be particularly useful for automated environments.
+
+## 4. Example Use-case
+Considering the nginx-ingres module, on typing the command 
+
+```
+kcl mod add https://github.com/kcl-lang/modules/tree/main/nginx-ingress /restrict-ingress-anotations
+``` 
+
+The following command will lead to the addition of restrict-ingress-anotations package to the current KCL project and will update the kcl.mod accordingly.
+
+Alternatively, for non-interactive environments, users can directly edit the `kcl.mod` file:
+```
+[package]
+name = "test"
+edition = "v0.9.0"
+version = "0.0.1"
+
+[dependencies]
+my_dependency = { git = "https://github.com/example/repo", subdirectory = "path/to/package" }
```

Comment:
The problem here is the same as above.

```
[dependencies]
my_dependency = { git = "https://github.com/example/repo/path/to/package" }
```

Any problem like this ?

Reply:
made the changes